### ISSUE TITLE
Binder build error for JupyterLab extension

### ISSUE THREAD

**Issue by jrbourbeau** [manics/jupyter-offlinenotebook#28]
When attempting to test out the JupyterLab extension via the [binder link](https://mybinder.org/v2/gh/manics/jupyter-offlinenotebook/master?urlpath=lab%2Ftree%2Fexample.ipynb) in the README I get the following error during the binder build:

```
info No lockfile found.
[1/4] Resolving packages...
warning @jupyterlab/application > @jupyterlab/ui-components > @blueprintjs/core > popper.js@1.16.1: You can find the new Popper v2at @popperjs/core, this package is dedicated to the legacy v1
warning @jupyterlab/application > @jupyterlab/ui-components > @blueprintjs/core > react-popper > popper.js@1.16.1: You can find the new Popper v2 at @popperjs/core, this package is dedicated to the legacy v1
[2/4] Fetching packages...
warning "@jupyterlab/application > @jupyterlab/ui-components@1.2.1" has unmet peer dependency "react@~16.8.4".
warning "@jupyterlab/application > @jupyterlab/ui-components > @blueprintjs/core@3.24.0" has unmet peer dependency "react-dom@^15.3.0 || 16".
warning "@jupyterlab/application > @jupyterlab/ui-components > @blueprintjs/select@3.12.0" has unmet peer dependency "react-dom@^15.3.0 || 16".
warning "@jupyterlab/application > @jupyterlab/ui-components > @blueprintjs/core > react-transition-group@2.9.0" has unmet peer dependency "react-dom@>=15.0.0".
[3/4] Linking dependencies...
[4/4] Building fresh packages...
success Saved lockfile.
$ jlpm run clean && jlpm run build
yarn run v1.15.2
$ rimraf lib tsconfig.tsbuildinfo
Done in 0.24s.
yarn run v1.15.2
$ tsc
Done in 6.32s.
Done in 16.20s.
+ jupyter labextension install
Building jupyterlab assets (build:dev:minimize)
An error occured.
RuntimeError: npm dependencies failed to install
See the log file for details:  /tmp/jupyterlab-debug-otczn4iq.log
Removing intermediate container 18137a6011f1
The command '/bin/sh -c ./binder/postBuild' returned a non-zero code: 1
```

**Comment by manics** [comment]
Thanks for opening the issue. I think it might be because it's also included in repo2docker https://github.com/jupyter/repo2docker/pull/845 and I suspect there's a conflict. I need to update this extension for JupyterLab 2.0 so I'll look into this error at the same time.

**Comment by jrbourbeau** [comment]
Thanks for the response @manics! I actually just came across that PR and am trying to build a different project with a previous `repo2docker` release that doesn't include `jupyter-offlinenotebook` (will post here if that helps resolve existing build issues) 

> I need to update this extension for JupyterLab 2.0 so I'll look into this error at the same time

Awesome, thank you!

**Comment by manics** [comment]
I fixed the binder build in #29 
https://github.com/manics/jupyter-offlinenotebook/pull/30 is the PR for JupyterLab 2 support, currently icons aren't displayed.

**Comment by jrbourbeau** [comment]
Great, thanks @manics!
